src/Knp/Menu/MenuItem.php

```diff
@@ -354,7 +354,7 @@ public function reorderChildren(array $order): ItemInterface
 
         $newChildren = [];
 
-        /** @var string $name */
+        /** @var string[] $order */
```

Comment:
put that in the phpdoc of the method instead, which documents the signature for the outside too

Reply:
That's a problem, since method is already signed (in interface) as `array<int|string, string|self>` and here we're using array elements as string.
So, we need to change signature to `array<int|string, string>` or change some code here

Reply:
If the implementation does not accept passing item instances, the interface should not allow it either. Otherwise, the implementation does not respect the interface.